### PR TITLE
Initial implementation of `syn-teams.libsonnet`

### PR DESCRIPTION
**Pull request by simu**
## TODO

- [x] Add PR template
- [x] Add some linting & testing infrastructure

### CODE REVIEW

**Comment by simu** [comment]
Boilerplate is copied from component template. Testing infrastructure is adapted from commodore-compile-pipelines.

File: syn-teams.libsonnet [hunk 1]
```diff
@@ -0,0 +1,237 @@
...
+ * \brief A map from application instances to the assigned team
+ *
+ * This raises an error if any component instances that are defined in
+ * `inv.applications` aren't assigned to a team (and are explicitly removed in
+ * the owner team) or if any instances are assigned to multiple teams.
+ *
+ * \returns a map from instance names to the assigned team (including the team owning the cluster)
+ */
+local applicationTeamMap =
+  local appMap = std.foldl(
```
Comment (haasad) [inline syn-teams.libsonnet]: could maybe be more easily understandable in an object comprehension

File: syn-teams.libsonnet [hunk 2]
```diff
@@ -0,0 +1,237 @@
...
+ * Raises an error if the cluster contains instances that aren't assigned to
+ * any team (taking into account the syn.owner fallback assignment) or if any
+ * instances are owned by multiple teams.
+ *
+ * \arg app the instance name to lookup in the map
+ * \arg default the default team to return when the instance isn't present in the map. Defaults to `null`
+ *
+ * \returns a list of applications assigned to the team or the value of `default` if the application doesn't exist in the map.
+ */
+local teamForApplication(app, default=null) = std.get(applicationTeamMap, app, default);
```
Comment (haasad) [inline syn-teams.libsonnet]:
```suggestion
local teamForApplication(app) = std.get(applicationTeamMap, app);
```

File: tests/syn-teams.jsonnet [hunk 3]
```diff
@@ -0,0 +1,124 @@
...
+  local ks2 = syn_teams.appKeys('foo-bar');
+  assert std.length(ks2) == 1 && ks2[0] == 'foo_bar' : 'appKeys should transform - to _';
+  local ks3 = syn_teams.appKeys('foo-bar as bar');
+  assert std.length(ks3) == 2 && ks3[0] == 'bar' && ks3[1] == 'foo_bar' : 'appKeys parses and transforms x-z as y';
+  local ks4 = syn_teams.appKeys('foo-bar as bar-qux');
+  assert std.length(ks4) == 2 && ks4[0] == 'bar_qux' && ks4[1] == 'foo_bar' : 'appKeys parses and transforms x-zl as y-w';
+  local ks5 = syn_teams.appKeys('foo-bar as bar-qux', raw=true);
+  assert std.length(ks5) == 2 && ks5[0] == 'bar-qux' && ks5[1] == 'foo-bar' : "appKeys(raw=true) doesn't transform - to _";
+  };
+  */
```
Comment (haasad) [inline tests/syn-teams.jsonnet]:
```suggestion
```

**Review by haasad — APPROVED** [review]

**Pull request merged**